waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -34,3 +39,91 @@
+                                                        :verbose true
+                                                        :service-id service-id)
+                                  (reduce set/union))]
+            (is (> (count instance-ids) 1) (str instance-ids)))
+
+          (log/info "waiting for at least one instance to get killed")
+          (is (wait-for #(->> (get-in (service-settings waiter-url service-id) [:instances :killed-instances])
+                              (map :id)
+                              set
+                              seq)
+                        :interval 2 :timeout 45)
+              (str "No killed instances found for " service-id))
+
+          ;; Test that the active instances' logs are available.
+          ;; This portion of the test logic was copied from basic-test/test-basic-logs
+          (let [active-instances (get-in (service-settings waiter-url service-id :cookies cookies)
+                                         [:instances :active-instances])
+                log-url (:log-url (first active-instances))
+                _ (log/debug "Log Url Active:" log-url)
+                make-request-fn (fn [url] (make-request url "" :verbose true))
+                {:keys [body] :as logs-response} (make-request-fn log-url)
+                _ (assert-response-status logs-response 200)
+                _ (log/debug "Response body:" body)
+                log-files-list (walk/keywordize-keys (json/read-str body))
+                stdout-file-link (:url (first (filter #(= (:name %) "stdout") log-files-list)))
+                stderr-file-link (:url (first (filter #(= (:name %) "stderr") log-files-list)))]
+            (is (every? #(string/includes? body %) ["stderr" "stdout"])
+                (str "Directory listing is missing entries: stderr and stdout, got response: " logs-response))
+            (doseq [file-link [stderr-file-link stdout-file-link]]
+              (if (string/starts-with? (str file-link) "http")
+                (assert-response-status (make-request-fn file-link) 200)
+                (log/warn "test-basic-logs did not verify file link:" stdout-file-link))))
+
+          ;; Test that the killed instances' logs were persisted to S3.
+          ;; This portion of the test logic was modified from the active-instances tests above.
+          (let [log-bucket-url (k8s-log-bucket-url waiter-url)
+                killed-instances (get-in (service-settings waiter-url service-id :cookies cookies)
+                                         [:instances :killed-instances])
+                log-url (:log-url (first killed-instances))
+                make-request-fn (fn [url] (make-request url "" :verbose true))
+                _ (do
+                    (log/info "waiting s3 logs to appear")
+                    (is (wait-for
+                          #(let [{:keys [body] :as logs-response} (make-request-fn log-url)]
+                             (string/includes? body log-bucket-url))
```

Comment:
Since this is the S3 logs test, should we instead wait for the prefix to start with the S3 log url?